Sources/Engine/Entities/Entity.h

```diff
@@ -694,18 +694,16 @@ inline CEntity& CEntityPointer::operator*(void) const { return *ep_pen; }
 /////////////////////////////////////////////////////////////////////
 // Reference counting functions
 inline void CEntity::AddReference(void) { 
-  if (this!=NULL) {
-    ASSERT(en_ctReferences>=0);
-    en_ctReferences++; 
-  }
+  ASSERT(this!=NULL);
```

Comment:
I've definitely seen this one come through as NULL before. Are these not asserting for you?